go/teams/seitan_v2_test.go

```diff
@@ -509,3 +511,173 @@ func TestTeamInviteSeitanV2Failures(t *testing.T) {
+			Akey:        keybase1.SeitanAKey(seitanRet.encoded),
+			Role:        keybase1.TeamRole_WRITER,
+			UnixCTime:   int64(timeNow),
+		}},
+	}
+	err = HandleTeamSeitan(context.Background(), tc.G, msg)
+	require.NoError(t, err)
+
+	// HandleTeamSeitan should not have added an invite for user. If it has, it
+	// also hasn't completed invite properly (`team.invite` link can't complete
+	// invite), which means the invite has been used but left active.
+	team, err := Load(context.TODO(), tc.G, keybase1.LoadTeamArg{
+		Name:        teamName.String(),
+		NeedAdmin:   true,
+		ForceRepoll: true,
+	})
+	require.NoError(t, err)
+
+	invite, _, found := team.FindActiveKeybaseInvite(user.GetUID())
+	require.False(t, found, "Expected not to find invite for user: %s", spew.Sdump(invite))
+}
+
+func TestSeitanMultipleRequestForOneInvite(t *testing.T) {
+	// Test server sending a Seitan notifications with multiple request for one
+	// Seitan invite. Seitan V1/V2 can never be multiple use, so at most one
+	// request should be handled.
+
+	tc := SetupTest(t, "team", 1)
+	defer tc.Cleanup()
+
+	tc.Tp.SkipSendingSystemChatMessages = true
+
+	admin, err := kbtest.CreateAndSignupFakeUser("team", tc.G)
+	require.NoError(t, err)
+
+	teamName, teamID := createTeam2(tc)
+
+	token, err := CreateSeitanTokenV2(context.Background(), tc.G,
+		teamName.String(), keybase1.TeamRole_WRITER, keybase1.SeitanKeyLabel{})
+	require.NoError(t, err)
+
+	// Create two users
+	var users [2]*kbtest.FakeUser
+	for i := range users {
+		kbtest.Logout(tc)
+
+		user, err := kbtest.CreateAndSignupFakeUser("team", tc.G)
+		require.NoError(t, err)
+		users[i] = user
+	}
+
+	timeNow := keybase1.ToTime(tc.G.Clock().Now())
+
+	var acceptances [2]acceptedSeitanV2
+	for i, user := range users {
+		kbtest.LogoutAndLoginAs(tc, user)
+		seitanRet, err := generateAcceptanceSeitanV2(SeitanIKeyV2(token), user.GetUserVersion(), timeNow)
+		require.NoError(t, err)
+		acceptances[i] = seitanRet
+
+		if i == 0 {
+			// First user has to PostSeitan so invite is changed to ACCEPTED on
+			// the server.
+			err = postSeitanV2(tc.MetaContext(), seitanRet)
+			require.NoError(t, err)
+		}
+	}
+
+	kbtest.LogoutAndLoginAs(tc, admin)
+
+	inviteID, err := acceptances[0].inviteID.TeamInviteID()
+	require.NoError(t, err)
+
+	var seitans [2]keybase1.TeamSeitanRequest
+	for i, user := range users {
+		seitans[i] = keybase1.TeamSeitanRequest{
+			InviteID:    inviteID,
+			Uid:         user.GetUID(),
+			EldestSeqno: user.EldestSeqno,
+			Akey:        keybase1.SeitanAKey(acceptances[i].encoded),
+			Role:        keybase1.TeamRole_WRITER,
+			UnixCTime:   int64(timeNow),
+		}
+	}
+	msg := keybase1.TeamSeitanMsg{
+		TeamID:  teamID,
+		Seitans: seitans[:],
+	}
+	err = HandleTeamSeitan(context.Background(), tc.G, msg)
+	if err != nil {
+		// We are expecting no error, but if there's a specific bug that we can
+		// recognize, inform about it.
+		if strings.Contains(err.Error(), "failed to update invites for") {
```

Comment:
can we do an error type check here?

Reply:
in general I think we should migrate our code over, it makes things a lot more robust even in app code